actix-threadpool/src/lib.rs

```diff
@@ -70,19 +66,17 @@ where
 
 /// Blocking operation completion future. It resolves with results
 /// of blocking function execution.
-pub struct CpuFuture<I, E> {
-    rx: oneshot::Receiver<Result<I, E>>,
+pub struct CpuFuture<I> {
+    rx: oneshot::Receiver<I>,
 }
 
-impl<I, E: fmt::Debug> Future for CpuFuture<I, E> {
-    type Item = I;
-    type Error = BlockingError<E>;
+impl<I> Future for CpuFuture<I> {
+    type Output = Result<I,Cancelled>;
 
-    fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
-        let res = futures::try_ready!(self.rx.poll().map_err(|_| BlockingError::Canceled));
-        match res {
-            Ok(val) => Ok(Async::Ready(val)),
-            Err(err) => Err(BlockingError::Error(err)),
-        }
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let rx  = Pin::new(&mut Pin::get_mut(self).rx);
```

Comment:
```suggestion
        let rx = Pin::new(&mut Pin::get_mut(self).rx);
```

I think this is not affected by readability.